File: lib/devise_ldap_authenticatable/ldap/ldapter.rb
require 'java'
java_import com.unraveldata.auth.Ldapter

module Devise
  module LDAP
    module Ldapter
      
      def self.valid_credentials?(login, password_plaintext)
        Java::ComUnraveldataAuth::Ldapter.new.authenticate(login, password_plaintext)
      end
      
    end
  end
end

File: com/unraveldata/auth/Ldapter.java
package com.unraveldata.auth;

/**
 * LDAP Adapter.
 *
 * Created: 20160309T1857
 *
 * @author pbaclace
 */
public class Ldapter {

    public boolean authenticate(String user, String password) {
        //ToDo: real implementation using JNDI
        if ("utest".equals(user) && "fortytwo!".equals(password)) {
            return true;
        } else {
            return false;
        }
    }
}
